pyomo/contrib/doe/fim_doe.py

```diff
@@ -0,0 +1,2582 @@
+                f.close()
+
+            # calculate jacobian
+            jac = self.__finite_calculation(output_record, scena_gen)
+
+            time11 = time.time()
+            self.logger.info('Build time with sequential_finite mode [s]:', sum(time_allbuild))
+            self.logger.info('Solve time with sequential_finite mode [s]:', sum(time_allsolve))
+            self.logger.info('Total wall clock time [s]:', time11-time00)
+
+            # return all models formed
+            self.models = models
+
+        # Assemble and analyze results
+        if self.specified_prior is None:
+            prior_in_use = self.prior_FIM
+        else:
+            prior_in_use = self.specified_prior
+
+        FIM_analysis = FIM_result(self.param_name, self.measure, jacobian_info=None, all_jacobian_info=jac,
+                                    prior_FIM=prior_in_use, store_FIM=self.FIM_store_name, scale_constant_value=self.scale_constant_value)
+
+        # Store the Jacobian information for access by users
+        self.jac = jac
+
+        if read_output is None:
+            FIM_analysis.build_time = sum(time_allbuild)
+            FIM_analysis.solve_time = sum(time_allsolve)
+
+        return FIM_analysis
+
+    def __sequential_sipopt(self,read_output):
+        time00 = time.time()
+        # create scenario class for a base case
+        scena_gen = Scenario_generator(self.param_init, formula=None, step=self.step)
+        scenario_all = scena_gen.simultaneous_scenario()
+
+        # sipopt only uses backward difference scheme
+        # store measurements for scenarios
+        all_perturb_measure = []
+        all_base_measure = []
+        # store jacobian info
+        jac={}
+
+        # if measurements are provided
+        if read_output is not None:
+            with open(read_output, 'rb') as f:
+                output_record = pickle.load(f)
+                f.close()
+            jac = self.__finite_calculation(output_record, scena_gen)
+
+        else:
+            # time building time and solving time store list
+            time_allbuild = []
+            time_allsolve = []
+            # loop over parameters
+            for pa in range(len(self.param_name)):
+                perturb_mea = []
+                base_mea = []
+
+                # create model
+                time0_build = time.time()
+                mod = self.create_model(scenario_all, self.args)
+                time1_build = time.time()
+                time_allbuild.append(time1_build - time0_build)
+
+                # discretize if needed
+                if self.discretize_model is not None:
+                    mod = self.discretize_model(mod)
+
+                # For sIPOPT, fix model DOF
+                if self.mode =='sequential_sipopt':
+                    mod = self.__fix_design(mod, self.design_values, fix_opt=True)
+
+                # extract (discretized) time
+                time_set = []
+                for t in mod.t:
```

Comment:
We should have the user pass in the name of the time set and not just assume/hard-code it to `t`.